### ISSUE TITLE
Is max-clustering-key-restrictions-per-query too low?

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla#10577]
In commit 080989d29624aeced6fab113ebdcbfa6e85be57b, @avikivity added a limit for the size of cartesian product of IN restrictions, because if it's too big it leads to the out-of-memory as described in #4752.

There's a separately configured maximum for partition keys and for clustering keys. Both currently default to 100 in `cql3/restrictions/restrictions_config.hh`, but can also be overridden by the `--max-partition-key-restrictions-per-query` and `--max-clustering-key-restrictions-per-query` configuration options. (see also issue #8882 about making these options part of "safe mode" and improving the error handling).

My question in this issue is whether maybe our default clustering-key limit of 100 is too low. For example. the translated Cassandra unit test `cassandra_tests/validation/operations/select_single_column_relation_test.py`::`testLargeClusteringINValues` does this simple query with an IN with 10,000 clustering keys:
```python
        inValues = list(range(10000))
        execute(cql, table, "SELECT * FROM %s WHERE k=? AND c IN ?", 0, inValues)
```
This request works fine on Cassandra but doesn't in Scylla, and I think we may be too strict. Unlike a IN on the partition key which may require Scylla to read a huge amount of data just to merge the results, in this case we are asked for 10,000 clustering keys, sequentially. I don't think this should cause out-of-memory errors like in #4752 (the examples there were with partition keys).

If we do increase the clustering-key limit beyond what fits in a single page, we should write a test that paging works correctly in this case. It's not very difficult, but not trivial either (the in values before the paging position should be ignored) so we should test that it is supported correctly.

**Comment by slivne** [comment]
I don't think our limits don;t make sense - user trip on this.

As a reference this is Astra's (the managed cassandra offering by Datasta limits) 
https://docs.datastax.com/en/astra/docs/plan/planning.html#_serverless_database_limits

**Comment by soyacz** [comment]
We face this issue when using Gemini in tests - limit is very low leading to many failures. See https://github.com/scylladb/gemini/issues/271

**Comment by yarongilor** [comment]
Reproduced in 5.0 as well, so we started using a workaround by setting Gemini parameter `--max-partition-keys 4 `  (instead of default 6).


https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/byo-longevity-test/182/ run failed.
SCT reported error:
```
AssertionError: [{'write_ops': 4882898, 'write_errors': 0, 'read_ops': 0, 'read_errors': 1, 'errors': [{'timestamp': '2022-08-04T17:25:16.552520061Z', 'message': 'Validation failed: unable to load check data from the test store: system failed: clustering-key cartesian product size 243 is greater than maximum 100', 'query': 'SELECT * FROM ks1.table1 WHERE pk0 IN (104,78,49) AND pk1 IN (8903584123377872509,3877567267038065175,61784911708610623) AND pk2 IN (-7376,-3756,22238) AND pk3 IN (4668058374148660249,7087430390359095663,1970106901461334675) AND pk4 IN (29096,29403,24177) '}]}]
```

**Comment by nyh** [comment]
@yarongilor note that if you want, you can also work around this by configuring Scylla (instead of Gemini), giving it a higher `--max-clustering-key-restrictions-per-query`.

**Comment by yarongilor** [comment]
> @yarongilor note that if you want, you can also work around this by configuring Scylla (instead of Gemini), giving it a higher `--max-clustering-key-restrictions-per-query`.

Right, if it sounds reasonable, SCT can configure it to Cassandra's default (or higher if needed for Gemini, i'm not sure what's the right number for it. can i just set it to 1000? )

**Comment by Mokto** [comment]
In the meantime, what is the best way of getting 10000 (for example) random rows in scylladb, just knowing their partition key?
